src/rgw/rgw_rados.cc

```diff
@@ -8741,6 +8741,8 @@ int RGWRados::get_obj_state_impl(RGWObjectCtx *rctx, const RGWBucketInfo& bucket
       s->manifest.set_head(bucket_info.placement_rule, obj, s->size); /* patch manifest to reflect the head we just read, some manifests might be
                                              broken due to old bugs */
       s->size = s->manifest.get_obj_size();
+      if (s->accounted_size < s->size)
```

Comment:
you're right that this needs to be conditional, but i'd like to be more explicit about the condition being `if (!compressed)`. can you add a flag above for that? something like:
```diff
   s->accounted_size = s->size;
 
   auto iter = s->attrset.find(RGW_ATTR_COMPRESSION);
-  if (iter != s->attrset.end()) {
+  const bool compressed = iter != s->attrset.end();
+  if (compressed) {
     // use uncompressed size for accounted_size
     try {
```